### PR TITLE
Moved WindowsFormsIntegration area

### PR DESCRIPTION
**Pull request by anjali-wpf**
[Fixes Issue ](issue: [ Migrate ] WindowsFormsIntegration Area - Feature Tests) #67 

## Description

Adds the WindowsFormsIntegration Area - Feature Tests

## Customer Impact

Developers can run the tests using .\RunTests.cmd /Area=WindowsFormsIntegration command and test their fixes.

## Testing

Built and ran the tests.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/wpf-test/pull/230)

### CODE REVIEW

File: src/Test/WindowsFormsIntegration/FeatureTests/ElementHost/BubbledEvents/BubbledEvents.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        }
+        return base.BeforeScenario(p, scenario);
+    }
+
+    void avControl_KeyDown(object sender, System.Windows.Input.KeyEventArgs e)
+    {
+        //assign RoutingStrategy to global string so a given scenario can examine it.
+        //RoutingStrategy += ((System.Windows.FrameworkElement)sender).Name + ":" + e.RoutedEvent.RoutingStrategy.ToString() + "::";
+
+        //Changed 4/27/06 per Nathan Enright to only look at the routing strategy itself:
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/ElementHost/BubbledEvents/BubbledEvents.cs:103]: @anjali-wpf Occurrence of date and name.

Comment (anjali-wpf) [reply src/Test/WindowsFormsIntegration/FeatureTests/ElementHost/BubbledEvents/BubbledEvents.cs:103]: Addressed

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/GenericControlTestBase.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                        case "RADIOBUTTON" :        TestedControl = new RadioButton(); break;
+                        case "RICHTEXTBOX" :        TestedControl = new RichTextBox(); break;
+                        case "SCROLLABLECONTROL" :  TestedControl = new ScrollableControl(); break;
+                        case "SPLITTER" :           TestedControl = new Splitter(); break;
+                        case "TABCONTROL" :         TestedControl = new TabControl(); break;
+                        case "TRACKBAR" :           TestedControl = new TrackBar(); break;
+                        case "TREEVIEW" :           TestedControl = new TreeView(); break;
+                        case "VSCROLLBAR" :         TestedControl = new VScrollBar(); break;
+
+                        /* KevinTao: Added 3/25/04
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/GenericControlTestBase.cs:84]: @anjali-wpf Occurrence of name and date.

File: src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/XWindowsFormsHost/WPFAutoPME.cs [hunk 3]
```diff
@@ -0,0 +1,1180 @@
...
+                else
+                    scenarioParams.log.WriteLine("SECURITY: Ignoring stack check.  Reason: \"" + reason + "\"");
+            }
+
+            BeginSecurityCheck(p);
+            _ignoreStackForCurrentScenario = ignoreStack;
+        }
+
+        //
+        // (12/10/03) We're currently grandfathering tests which use AddRequiredPermissions()
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/XWindowsFormsHost/WPFAutoPME.cs:692]: @anjali-wpf reference to a task and date.

File: src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/XWindowsFormsHost/WPFAutoPME.cs [hunk 4]
```diff
@@ -0,0 +1,1180 @@
...
+        // </param>
+        // <param term="expected">
+        //  The list of scenarios that are expected to be implemented by the testcase
+        // </param>
+        // <seealso member="missingScenariosInGroup"/>
+        // <seealso member="MatchScenarioToMethod"/>
+        // </doc>
+        private void InitMissingScenariosForGroup(ScenarioGroup key, MethodInfo[] scenarios, ArrayList expected)
+        {
+            // KEVINTAO 10/17/03: Modified to search the entire scenario list for an expected test.
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/XWindowsFormsHost/WPFAutoPME.cs:918]: @anjali-wpf presence of an alias and date

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs [hunk 5]
```diff
@@ -0,0 +1,930 @@
...
+        {
+            if (IsHackValid(failAfterVersion))
+            { log.WriteLine(GetHackComments(hackComments, failAfterVersion)); }
+        }
+
+		/// <summary>
+		/// Constructs a new Scenario Result. If the Hack has expired it is considered a Fail else it 
+		/// it considered Pass. Appropriate comments are added to the log file. In case of Failure the known
+		/// 
+
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs:245]: @anjali-wpf  inconsistent line breaks.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs [hunk 6]
```diff
@@ -0,0 +1,930 @@
...
+            if (IsHackValid(failAfterVersion))
+            { return string.Format("Hack will after version: {0}", failAfterVersion);  }
+            else
+            { return string.Format("Hack has expired after version: {0}", failAfterVersion); }
+        }
+
+		/// <summary>
+		/// Gets comments for the hack along with the information on the expiration
+		/// of the hack and the 
+
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs:349]: @anjali-wpf  inconsistent line breaks here as well.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs [hunk 7]
```diff
@@ -0,0 +1,930 @@
...
+		/// <summary>
+		///  Reports failure if hack has expired. Increments the pass/fail counters.  
+		/// If this increments the fail counter, the comments will be printed to 
+		/// the given Log object.
+		/// </summary>
+		/// <param name="log">
+		///  If this is not null, comments and details are printed to the log.
+		/// </param>
+		/// <param name="db">
+		///
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Log/ScenarioResult.cs:753]: @anjali-wpf  inconsistent line breaks.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs [hunk 8]
```diff
@@ -0,0 +1,1121 @@
...
+				else
+					scenarioParams.log.WriteLine("SECURITY: Ignoring stack check.  Reason: \"" + reason + "\"");
+			}
+
+			BeginSecurityCheck(p);
+			_ignoreStackForCurrentScenario = ignoreStack;
+		}
+
+		//
+		// (12/10/03) We're currently grandfathering tests which use AddRequiredPermissions()
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs:653]: @anjali-wpf mention of date and task.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs [hunk 9]
```diff
@@ -0,0 +1,1121 @@
...
+        // </param>
+        // <param term="expected">
+        //  The list of scenarios that are expected to be implemented by the testcase
+        // </param>
+        // <seealso member="missingScenariosInGroup"/>
+        // <seealso member="MatchScenarioToMethod"/>
+        // </doc>
+        private void InitMissingScenariosForGroup(ScenarioGroup key, MethodInfo[] scenarios, ArrayList expected)
+        {
+			// KEVINTAO 10/17/03: Modified to search the entire scenario list for an expected test.
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs:864]: @anjali-wpf  alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs [hunk 10]
```diff
@@ -0,0 +1,1121 @@
...
+			//    This essentially eliminates the need for OverrideScenario.  Scott and I couldn't
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoPME.cs:865]: @anjali-wpf  name and scenario present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoTest.cs [hunk 11]
```diff
@@ -0,0 +1,864 @@
...
+					}
+				}
+
+				Debug.Assert((onMethod != null && customRaiseMethod == null) ||
+							 (onMethod == null && customRaiseMethod != null),
+							  "onMethod or customRaiseMethod must be non-null and the other must be null");
+
+				// 2. Has to have a delegate method defined
+				//
+				// KevinTao 10/24/01:
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoTest.cs:684]: @anjali-wpf alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoTest.cs [hunk 12]
```diff
@@ -0,0 +1,864 @@
...
+		protected override void InitTest(TParams p)
+		{
+			base.InitTest(p);
+			_properties = Class.GetProperties();
+			_events = Class.GetEvents();
+		}
+
+
+		//
+		// (KevinTao 9/13/02)
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/AutoTest.cs:79]: @anjali-wpf  alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs [hunk 13]
```diff
@@ -0,0 +1,5252 @@
...
+                }, typeof(Control).GetMethod("DrawToBitmap"),
+                LibSecurity.AllWindows);
+
+            if(bSet)
+                result.IncCounters(!Utilities.BitmapsIdentical(mapA, mapB), "nothing was drawn", p.log);
+
+            return result;
+        }
+
+        // KEVINTAO: ShouldSerialize methods are now internal.  I'll keep this code around for
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs:4716]: @anjali-wpf  alias present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs [hunk 14]
```diff
@@ -0,0 +1,5252 @@
...
+        }
+
+        protected virtual ScenarioResult get_Region(TParams p)
+        {
+            if ((_c = GetControl(p)) == null) return ScenarioResult.Fail;
+
+            Region orig = _c.Region;
+            Region region = p.ru.GetRegion(_c.Size);
+
+            //willsad 3/2/00
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs:2592]: @anjali-wpf  occurrence of alias and date.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs [hunk 15]
```diff
@@ -0,0 +1,5252 @@
...
+
+            // We'll use ContainsFocus instead of Focused since a control's child can have focus
+            // and Focused will return false
+            bool result = (canFocus && _c.ContainsFocus) || (!canFocus && !_c.ContainsFocus);
+
+            p.log.WriteLine("after Select Focused = " + _c.Focused.ToString());
+            return new ScenarioResult(result);
+        }
+
+        //todo: may want to make this test against form or a container control and verify that the next control is selected(activated)
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs:2145]: @anjali-wpf  TODO  present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs [hunk 16]
```diff
@@ -0,0 +1,5252 @@
...
+        {
+            // These used to use "if ( c.GetType == tyepof(Foo) )", but these wouldn't cover
+            // subclasses for inheritance tests, so we changed them to use the "is" operator
+            if (c is DateTimePicker) return true;
+
+            if (c is MonthCalendar) return true;
+
+            if (c is GroupBox) return true;
+
+            // KevinTao: These are UserPaint controls.  They should receive paint events.
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs:1410]: @anjali-wpf  alias present

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs [hunk 17]
```diff
@@ -0,0 +1,5252 @@
...
+            textInASCII(ss);
+            _c.Text = ss;
+
+            String s = _c.Text;
+
+            p.log.WriteLine("Text is " + s);
+            textInASCII(s);
+
+            // On Win9x, there are some chars that are not round-trippable even though
+            // they are valid Unicode chracters.  According to TadaoM, it is sufficient to
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/AutoPME/XControl.cs:820]: @anjali-wpf  alias present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/ReflectBase.cs [hunk 18]
```diff
@@ -0,0 +1,2441 @@
...
+                return Size.Empty;
+
+            if ( type == typeof(RectangleF) )
+                return RectangleF.Empty;
+
+            if ( type == typeof(SizeF) )
+                return SizeF.Empty;
+
+            if ( type.IsPrimitive ) {
+                // KevinTao 1/28/00: Yuck.  The primitives no longer have parameterless
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/ReflectTools/ReflectBase.cs:1431]: @anjali-wpf alias and date present. Do we need to change the comment as well?

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/InteropUtil.cs [hunk 19]
```diff
@@ -0,0 +1,435 @@
...
+            return new RECT(x,
+                            y,
+                            x + width,
+                            x + height);
+        }
+    }
+
+	//
+	//  Added to support new SystemInformation members for Whidbey TopAPI feature set.
+	//  antonioa - 2/12/2003
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/InteropUtil.cs:387]: @anjali-wpf alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/InteropUtil.cs [hunk 20]
```diff
@@ -0,0 +1,435 @@
...
+		public byte lfClipPrecision;
+		public byte lfQuality;
+		public byte lfPitchAndFamily;
+		[MarshalAs(UnmanagedType.ByValTStr, SizeConst=32)]
+		public string   lfFaceName;
+	}
+
+	//
+	//  Added to support new SystemInformation members for Whidbey TopAPI feature set.
+	//  antonioa - 2/12/2003
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/InteropUtil.cs:410]: @anjali-wpf alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/IntlString.cs [hunk 21]
```diff
@@ -0,0 +1,381 @@
...
+        //  A unicode character in the range for the required codepage that
+        //  is optionally printable.
+        // </retvalue>
+        // </doc>
+        public virtual char GetChar(bool bPrintableOnly)
+        {
+            int nRange       = this.GetRange(0, _lUnicodeCharRanges.Count - 1);
+            ByteRange range  = (ByteRange)(_lUnicodeCharRanges[nRange]);
+
+            //Vinay added this code to make it work with Win95 & Win98
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/IntlString.cs:220]: @anjali-wpf name present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/RandomUtil.cs [hunk 22]
```diff
@@ -0,0 +1,3465 @@
...
+            return retVal;
+        }
+
+        //
+        // In some locales which use different calendars (e.g. Arabic, which uses the Hijiri calendar),
+        // the range of valid DateTimes is smaller than the standard Gregorian calendar.  This method
+        // will return a DateTime that should be valid for the current culture.
+        //
+        // HACK: In v1, there is no direct way to query a calendar for its min and max dates.  So we
+        //       determine the min date by creating a DateTime for 1/1/0001 in the current culture's
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/RandomUtil.cs:1535]: @anjali-wpf can we confirm if this hack is to be mentioned and if the date is required.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/RandomUtil.cs [hunk 23]
```diff
@@ -0,0 +1,3465 @@
...
+        //  The maximum number of tries to get a Font before giving it up as a bad job
+        // </desc>
+        // <seealso member="GetFont"/>
+        // </doc>
+        private const int maxAttemptsToGetFont = 100;
+
+        // <doc>
+        // <desc>
+        //  The Random number generator
+        //  KevinTao 4/12/00: Made it static so all instances share the same
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/RandomUtil.cs:62]: @anjali-wpf  alias and date present.

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/WPFReflectBase/WPFReflectBase.cs [hunk 24]
```diff
@@ -0,0 +1,2577 @@
...
+
+            //if (type == typeof(RectangleF))
+            //    return RectangleF.Empty;
+
+            //if (type == typeof(SizeF))
+            //    return SizeF.Empty;
+
+            if (type.IsPrimitive)
+            {
+                // KevinTao 1/28/00: Yuck.  The primitives no longer have parameterless
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/WPFReflectBase/WPFReflectBase.cs:1515]: @anjali-wpf  Alias, date present. Can slang be removed?

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/XmlLogTree/BugInfoCollection.cs [hunk 25]
```diff
@@ -0,0 +1,233 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//    Information Contained Herein is Proprietary and Confidential.
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/XmlLogTree/BugInfoCollection.cs:5]: @anjali-wpf can we confirm if this header is needed?

File: src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/XmlLogTree/ScenarioCollection.cs [hunk 26]
```diff
@@ -0,0 +1,242 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//    Information Contained Herein is Proprietary and Confidential.
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/XmlLogTree/ScenarioCollection.cs:5]: @anjali-wpf header confirmation?

File: src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/DevDivBugs95607/DevDivBugs95607.cs [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+using WPFReflectTools;
+using System.Threading;
+using SW = System.Windows;
+using SWC = System.Windows.Controls;
+using System.Windows.Forms.Integration;
+using SWF = System.Windows.Forms;
+using System.Windows.Media;
+
+
+/// <Testcase>DevDivBugs95607</Testcase>
```
Comment (harshit7962) [inline src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/DevDivBugs95607/DevDivBugs95607.cs:18]: @anjali-wpf mention of devdiv bug id.

Comment (anjali-wpf) [reply src/Test/WindowsFormsIntegration/FeatureTests/WindowsFormsHost/DevDivBugs95607/DevDivBugs95607.cs]: we will take this separately

**Pull request merged**